src/components/Avatar/Avatar.jsx

```diff
@@ -0,0 +1,28 @@
+import { useRef } from 'react'
+import UserImage from '/src/components/UserImage'
+import * as Styled from './Avatar.styled'
+
+const Avatar = ({ user, onUpdateAvatar, imageKey }) => {
+
+  const fileInput = useRef(null)
+  const handleInputChange = (e) => {
+    e.preventDefault()
+    if (!e.target.files || !e.target.files[0]) return
+    onUpdateAvatar(e.target.files[0])
+  }
+
+  return (
+    <Styled.Avatar>
+        <input type="file" ref={fileInput} style={{ display: 'none' }}  multiple={false} onChange={handleInputChange} accept=".png, .jpeg, .jpg" />
+        <Styled.ImageIcon>
+          <Styled.AvatarIcon onClick={() => fileInput.current.click()} icon='edit' />
+          <UserImage size={100} name={user.name} imageKey={imageKey} />
+        </Styled.ImageIcon>
+        <Styled.Username>
```

Comment:
I'm not sure it makes sense to put the name in Avatar as it would make reusing the component difficult? Probably better in ProfilePage.jsx

Reply:
Done.